### PR TITLE
[mdatagen] mdatagen support validate metrics in in telemetry section

### PR DESCRIPTION
**Pull request by Frapschen**
#### Description

I have a use case, I define an attribute and try to use it in `telemetry.metric` however I encounter an error: `unused attributes: [xxx]`
So, I push this PR to enhance the `validateMetrics()` func to  support validate metrics in `telemetry.metric`

#### Link to tracking issue fix https://github.com/open-telemetry/opentelemetry-collector/issues/10925
Fixes #

### CODE REVIEW

File: cmd/mdatagen/validate.go [hunk 1]
```diff
@@ -129,34 +129,9 @@ func (md *metadata) validateResourceAttributes() error {
...
-				unknownAttrs = append(unknownAttrs, attr)
-			}
-		}
-		if len(unknownAttrs) > 0 {
-			errs = errors.Join(errs, fmt.Errorf(`metric "%v" refers to undefined attributes: %v`, mn, unknownAttrs))
-		}
-	}
-	errs = errors.Join(errs, md.validateAttributes(usedAttrs))
+	errs = errors.Join(errs, validateMetrics(md.Metrics, md.Attributes, usedAttrs),
+		validateMetrics(md.Telemetry.Metrics, md.Attributes, usedAttrs),
```
Comment (codeboten) [inline cmd/mdatagen/validate.go:133]: Thanks @Frapschen, is there already a unit test that covers this change?

Comment (Frapschen) [reply cmd/mdatagen/validate.go:133]: Yes, please see the changed file `cmd/mdatagen/testdata/unused_attribute.yaml`, it will make sure the attributes that are used in `telemetry.metric` will not encounter `unused attributes: [xxx]` error.

**Comment by Frapschen** [comment]
Hi @codeboten, PTAL thanks!

File: cmd/mdatagen/loader.go [hunk 2]
```diff
@@ -122,7 +122,7 @@ type metric struct {
 	Sum *sum `mapstructure:"sum,omitempty"`
 	// Gauge stores metadata for gauge metric type
 	Gauge *gauge `mapstructure:"gauge,omitempty"`
-	// Gauge stores metadata for gauge metric type
+	// Histogram stores metadata for histogram metric type
```
Comment (atoulme) [inline cmd/mdatagen/loader.go:125]: Split to its own PR

Comment (Frapschen) [reply cmd/mdatagen/loader.go]: PTAL https://github.com/open-telemetry/opentelemetry-collector/pull/11057

**Comment by Frapschen** [comment]
@codeboten just a friendly ping. Please review again.

**Review by codeboten — APPROVED** [review]
